contracts/adapters/PrtFeeSplitExtension.sol

```diff
@@ -0,0 +1,224 @@
+
+    event AnyoneAccrueUpdated(bool isAnyoneAllowedToBid);
+    event AccruerStatusUpdated(address indexed accruer, bool isBidderAllowed);
+    event OperatorFeeSplitUpdated(uint256 newFeeSplit);
+    event PrtFeesDistributed(
+        address indexed operatorFeeRecipient,
+        address indexed prtStakingPool,
+        uint256 operatorTake,
+        uint256 prtTake
+    );
+    event PrtStakingPoolUpdated(address newPrtStakingPool);
+
+    /* ============ Immutables ============ */
+
+    IPrt public immutable prt;
+
+    /* ============ State Variables ============ */
+
+    bool public isAnyoneAllowedToAccrue;
+    address[] accruersHistory;
+    mapping(address => bool) accrueAllowList;
+    IPrtStakingPool public prtStakingPool;
+
+    /* ============ Modifiers ============ */
+
+    modifier onlyAllowedAccruer() {
+        require(_isAllowedAccruer(msg.sender), "Not allowed to accrue");
+        _;
+    }
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IBaseManager _manager,
+        IStreamingFeeModule _streamingFeeModule,
+        IIssuanceModule _issuanceModule,
+        uint256 _operatorFeeSplit,
+        address _operatorFeeRecipient,
+        IPrt _prt
+    )
+        public
+        FeeSplitExtension(
+            _manager,
+            _streamingFeeModule,
+            _issuanceModule,
+            _operatorFeeSplit,
+            _operatorFeeRecipient
+        )
+    {
+        require(_prt.setToken() == address(manager.setToken()), "SetToken mismatch with Prt");
+        prt = _prt;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @notice MUTUAL UPGRADE: Updates PRT staking pool. PRT staking pool must have this extension set as the feeSplitExtension.
+     * @param _prtStakingPool Address of the new PRT staking pool
+     */
+    function updatePrtStakingPool(IPrtStakingPool _prtStakingPool) 
+        external 
+        mutualUpgrade(manager.operator(), manager.methodologist()) 
+    {
+        require(address(_prtStakingPool) != address(0), "Zero address not valid");
+        require(_prtStakingPool.feeSplitExtension() == address(this), "PrtFeeSplitExtension must be set");
+        prtStakingPool = _prtStakingPool;
+        emit PrtStakingPoolUpdated(address(_prtStakingPool));
+    }
+
+    /**
+     * @notice ONLY ALLOWED ACCRUER: Accrues fees from streaming fee module. Gets resulting balance after fee accrual, calculates fees for
+     * operator and PRT staking pool, and sends to operator fee recipient and PRT Staking Pool respectively. NOTE: mint/redeem fees
+     * will automatically be sent to this address so reading the balance of the SetToken in the contract after accrual is
+     * sufficient for accounting for all collected fees. If the PRT take is greater than 0, the PRT Staking Pool will accrue the fees
+     * and update the snapshot.
+     */
+    function accrueFeesAndDistribute() public override onlyAllowedAccruer {
```

Comment:
Maybe we should add a line for `require(address(prtStakingPool) != address(0), "PRT Staking Pool not set")`, as a reminder that this function can be called before the prtStakingPool is defined?

Reply:
added here [`e7a8a8c`](https://github.com/IndexCoop/index-coop-smart-contracts/pull/176/commits/e7a8a8c72e506446d2d42866201503ad2181a9c6)